pattlog.go

```diff
@@ -66,14 +82,24 @@ func FormatLogRecord(format string, rec *LogRecord) string {
 	for i, piece := range pieces {
 		if i > 0 && len(piece) > 0 {
 			switch piece[0] {
+			case 'A':
+				// Don't recompute this pattern unless it's actually used,
+				// string formatting is expensive 
+				if millisFormatCache.lastMillis != millis {
+					nano := rec.Created.Nanosecond()
+					millisString := fmt.Sprintf("%02d:%02d:%02d.%03d", hour, minute, second, nano / 1e6)
+					millisFormatCache.lastMillis = millis
+					millisFormatCache.millisTime = millisString
+				}
```

Comment:
nit: logic inside of case statements can make them unwieldy pretty quickly.  If I was choosing I'd say the cache invalidation/updating should happen outside like the second level cache.